rpc/send.go

```diff
@@ -154,121 +143,92 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 	// heartbeat measure ping times. With a bit of seasoning, each
 	// node will be able to order the healthy replicas based on latency.
 
-	replies := []interface{}(nil)
-	helperChan := make(chan interface{}, len(clients))
-	N := opts.N
-	errors := 0
-	retryableErrors := 0
-	successes := 0
-	index := 0
+	done := make(chan *rpc.Call, len(addrs)) // TRIM: maybe do fewer opts.N
 
-	// Send RPCs to replicas as necessary to achieve opts.N successes.
-	for {
-		// Start clients up to N.
-		for ; index < N; index++ {
-			args := getArgs(clients[index].Addr())
-			if args == nil {
-				helperChan <- util.Errorf("nil arguments returned for client %s", clients[index].Addr())
-				continue
-			}
-			reply := getReply()
+	sendFn := func(addr net.Addr) {
+		// NewClient might block while connecting, but connecting in parallel
+		// blows up the tests.
+		client, err := NewClient(addr, context)
+		if err != nil {
+			done <- &rpc.Call{Error: newRPCError(util.Errorf("connection to %s failed: %s", addr, err))}
+			return
+		}
+
+		if args := getArgs(addr); args != nil {
 			if log.V(2) {
-				log.Infof("%s: sending request to %s: %+v", method, clients[index].Addr(), args)
+				log.Infof("%s: sending request to %s: %+v", method, addr, args)
 			}
-
-			trace.Event(fmt.Sprintf("sending to %s", clients[index].Addr()))
-			go sendOneFn(clients[index], opts.Timeout, method, args, reply, helperChan)
+			trace.Event(fmt.Sprintf("sending to %s", addr))
+			go sendOneFn(client, opts.Timeout, method, args, getReply(), done)
+		} else {
+			done <- &rpc.Call{Error: newRPCError(util.Errorf("nil arguments returned for client %s", addr))}
 		}
-		// Wait for completions.
+	}
+
+	// Start clients up to opts.N.
```

Comment:
It looks like opts.N is always 1, and i'm not sure when we'd ever change that. Should we just rip out all the support for RPCs that expect multiple replies?